pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -59,14 +59,26 @@ type HostAddress struct {
 // See kube.ConvertService for the conversion from K8S to internal Service.
 func ServiceToServiceEntry(svc *model.Service, proxy *model.Proxy) *config.Config {
 	gvk := gvk.ServiceEntry
+	getSvcAddresses := func(s *model.Service, node *model.Proxy) []string {
+		if node.Metadata != nil && node.Metadata.ClusterID == "" {
+			var addresses []string
+			addressMap := s.ClusterVIPs.GetAddresses()
+			for _, clusterAddresses := range addressMap {
+				addresses = append(addresses, clusterAddresses...)
+			}
+			return addresses
+		}
+
+		return s.GetAllAddressesForProxy(proxy)
+	}
```

Comment:
I like removing `GetAddresses` in favor of this function. That one should never be exported.